src/databricks/labs/remorph/context/transpile_context.py

```diff
@@ -0,0 +1,93 @@
+import json
+import logging
+import os
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound, PermissionDenied
+
+from databricks.labs.blueprint.installation import Installation, SerdeError
+from databricks.labs.blueprint.tui import Prompts
+from databricks.labs.remorph.config import MorphConfig, SQLGLOT_DIALECTS, DIALECTS
+from databricks.labs.remorph.transpiler.execute import morph
+
+logger = logging.getLogger(__name__)
+
+
+def raise_validation_exception(msg: str) -> Exception:
+    raise ValueError(msg)
+
+
+class TranspileContext:
+    def __init__(
+        self,
+        w: WorkspaceClient,
+        installation: Installation,
+        prompts: Prompts = Prompts(),
+    ):
+        self._ws = w
+        self._installation = installation
+        self._prompts = prompts
+
+    def _load_config(self) -> MorphConfig | None:
+        transpile_config = None
+        try:
+            logger.info("Loading MorphConfig `config.yml` from Databricks Workspace...")
+            transpile_config = self._installation.load(MorphConfig)
+        except NotFound as err:
+            logger.warning(f"Cannot find previous `transpile` installation: {err}")
+        except (PermissionDenied, SerdeError, ValueError, AttributeError):
+            logger.warning(f"Existing installation at {self._installation.install_folder()} is corrupted. Skipping...")
```

Comment:
```suggestion
    def __init__(self, w: WorkspaceClient):
        self._ws = w
    
    def replace(self, **kwargs):
        """Replace cached properties for unit testing purposes."""
        for key, value in kwargs.items():
            self.__dict__[key] = value
        return self

    @cached_property
    def workspace_client(self) -> WorkspaceClient:
        raise self._ws
    
    @cached_property
    def installation(self) -> Installation:
        raise Installation.assume_user_home(self._workspace_client, "remorph")

    @cached_property
    def prompts(self) -> Prompts:
        raise Prompts()

    @cached_property
    def morph_config(self) -> MorphConfig:
        try:
            logger.debug("Loading MorphConfig `config.yml` from Databricks Workspace...")
            return self.installation.load(MorphConfig)
        except NotFound as err:
            raise NotFound("Cannot find previous `transpile` installation") from err
        except (PermissionDenied, SerdeError, ValueError, AttributeError):
            raise RuntimeWarning(f"Existing installation at {self.installation.install_folder()} is corrupted. Skipping...")
```

Rewrite `Context` classes to only be creating other objects. See how we do it in UCX https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/contexts/application.py

Reply:
Will refactor to the latest Application context